### PR TITLE
Makes nightmares unable to be pepersprayed

### PR DESCRIPTION
**Pull request by ghost**
## About The Pull Request

Adds a variable to eye organ that makes owner of this eyes pepperspray protected, and gives it to nightmare eyes.

## Why It's Good For The Game

Getting peperspray protection as a nightmare is kinda hard, since you can't wear gas mask. Also winning a horrific void creature just by peperspraying it sounds not really cool.

## Changelog
:cl:
balance: Nightmares no longer are able to be pepersprayed
/:cl:

### CODE REVIEW

**Comment by calliophis-maculiceps** [comment]
i mean it's funny as fuck

File: code/__DEFINES/traits.dm [hunk 1]
```diff
@@ -178,6 +178,7 @@ Remember to update _globalvars/traits.dm if you're adding/removing/renaming trai
 #define TRAIT_XENO_HOST "xeno_host"
 #define TRAIT_STUNIMMUNE "stun_immunity"
 #define TRAIT_BATON_RESISTANCE "baton_resistance"
+#define TRAIT_PEPERSPRAY_RESISTANCE "peperspray_resistance"
```
Comment (Fikou) [inline code/__DEFINES/traits.dm:181]:
```suggestion
#define TRAIT_PEPPERSPRAY_RESISTANCE "pepperspray_resistance"
```
you need to fix the other uses too

File: code/modules/mob/living/carbon/carbon_defense.dm [hunk 2]
```diff
@@ -45,6 +45,8 @@
 	return null
 
 /mob/living/carbon/is_pepper_proof(check_flags = ALL)
+	if(HAS_TRAIT(src, TRAIT_PEPERSPRAY_RESISTANCE))
+		return TRUE
```
Comment (Fikou) [inline code/modules/mob/living/carbon/carbon_defense.dm:49]: this should be on living side, not on carbon. make this proc call parent and then put this check on the parent proc. also this expects a ref, so i think you wanna return src

**Review by Mothblocks — CHANGES_REQUESTED** [review]
make it a feature of the eyes. makes no sense that nightmares themselves would be innately not pepper sprayable

**Comment by SuperSlayer0** [comment]
> make it a feature of the eyes. makes no sense that nightmares themselves would be innately not pepper sprayable

Okay

**Review by Mothblocks — APPROVED** [review]

**Pull request merged**